src/brogue/Monsters.c

```diff
@@ -714,6 +714,10 @@ boolean spawnMinions(short hordeID, creature *leader, boolean summoned) {
             if (hordeCatalog[hordeID].flags & HORDE_ALLIED_WITH_PLAYER) {
                 becomeAllyWith(monst);
             }
+            if (rogue.patchVersion >= 3 && (theHorde->flags & HORDE_MACHINE_THIEF)) {
+                monst->safetyMap = allocGrid(); // Keep thieves from fleeing before they see the player
+                fillGrid(leader->safetyMap, 0);
```

Comment:
you allocate monst's grid but then set leader's, is that intentional?

Reply:
You're right, I'll switch leader to monst real quick.